docs/excel/excel-add-ins-conditional-formatting.md

```diff
@@ -0,0 +1,305 @@
+conditionalFormat.cellValue.rule = { formula1: "=0", operator: "LessThan" };
+
+await context.sync();
+```
+
+### [Color scale](https://docs.microsoft.com/javascript/api/excel/excel.colorscaleconditionalformat)
+
+Color scale formatting applies a color gradient across the data range. The `criteria` property on the `ColorScaleConditionalFormat` defines the colors for the minimum, maximum, and an optional midpoint. Every cell will be proportionally colored based on its value relative to the minimum, midpoint, and maximum. The following example shows a range being colored blue to yellow to red.
+
+![A range with the low number in blue, average number in yellow, and high number is red, with gradients for between values.](../images/excel-conditional-format-color-scale.png)
+
+```typescript
+const sheet = context.workbook.worksheets.getItem("Sample");
+const range = sheet.getRange("B2:M5");
+const conditionalFormat = range.conditionalFormats.add(
+      Excel.ConditionalFormatType.colorScale
+);
+
+// color the backgrounds of the cells from blue to yellow to red based on value
+const criteria = {
+      minimum: {
+           formula: null,
+           type: Excel.ConditionalFormatColorCriterionType.lowestValue,
+           color: "blue"
+      },
+      midpoint: {
+           formula: "50",
+           type: Excel.ConditionalFormatColorCriterionType.percent,
+           color: "yellow"
+      },
+      maximum: {
+           formula: null,
+           type: Excel.ConditionalFormatColorCriterionType.highestValue,
+           color: "red"
+      }
+};
+conditionalFormat.colorScale.criteria = criteria;
+
+await context.sync();
+```
+
+### [Custom](https://docs.microsoft.com/javascript/api/excel/excel.customconditionalformat) 
+
+Custom conditional formatting applies a user-defined format to the cells based on a formula of arbitrary complexity. The following example colors the fonts green of cells with higher values than the cell to their left.
+
+![A range with green numbers for places the preceeding column's value in that row is lower.](../images/excel-conditional-format-custom.png)
+
+```typescript
+const sheet = context.workbook.worksheets.getItem("Sample");
+const range = sheet.getRange("B8:E13");
+const conditionalFormat = range.conditionalFormats.add(
+     Excel.ConditionalFormatType.custom
+);
+
+// set the fonts green of cells with higher values than the cell to their left
+conditionalFormat.custom.rule.formula = '=IF(B8>INDIRECT("RC[-1]",0),TRUE)';
+conditionalFormat.custom.format.font.color = "green";
+
+await context.sync();
+
+```
+### [Data bar](https://docs.microsoft.com/javascript/api/excel/excel.databarconditionalformat)
+
+Data bar formatting adds data bars to the cells, filled relative to the upper and lower bounds of the range. The `DataBarConditionalFormat` object has several properties to control the bar’s look. The following example formats the range with data bars filling left-to-right.
```

Comment:
The phrase "the upper and lower bounds of the range" is confusing . What are these bounds? What sets them.